### DISCUSSION TITLE
How can I translate game and play on Exult

### DISCUSSION THREAD

**Discussion by dfdgsdfg** [exult/exult#304] [General]
Is Exult have any CJK system in the engine?

If it has, how can I do that?

**Comment by dfdgsdfg** [comment]
Seems Exult does not support 2-byte language.

http://exult.info/forum/viewtopic.php?p=17696

**Comment by DominusExult** [comment]
Best take a look at what @wonst719 did for the Korean translation. Their fork makes it possible to use Korean symbols https://github.com/wonst719/exult (look at the branches).
The actual translation is another topic that maybe @ungesundes-halbwissen has the most up to date instructions (he translated the two Ultima VII games with their add-ons to German).

**Comment by r-h-o-n-a** [comment]
Afraid I can't say anything about doing a translation that requires CJK support. Though I'm currently working on a translation project for SI, all it needed was a custom font, and otherwise works from the basic ASCII codepoints.

However, regarding the translation itself, the shape (object) names are easy enough to deal with in Exult Studio. As for the usecode file (which includes texts from conversations, exclamations, books, signs, and so forth), that's a bit more involved. For my translation I'm using a little Windows utility called U7Editor 0.94, which was made in 2006 by Andrea Marzella for the team that did the Italian translation. It lets you translate text one line at a time, auto-saves changes into a series of .txt files, and then allows you to recompile these .txt files back into the original usecode. I can send you a copy of this utility if you like, as the original webpage is long since dead.

**Reply by DominusExult** [reply]
@r-h-o-n-a could you attach this utility here as well?

Which language are you translating to and is there a website/repository to follow its progress?

**Reply by ungesundes-halbwissen** [reply]
> For my translation I'm using a little Windows utility called U7Editor 0.94, which was made in 2006 by Andrea Marzella for the team that did the Italian translation. It lets you translate text one line at a time, auto-saves changes into a series of .txt files, and then allows you to recompile these .txt files back into the original usecode. I can send you a copy of this utility if you like, as the original webpage is long since dead.

Hi @r-h-o-n-a,
You can use v0.94 for U7BG alright, but please do not try to use it for SI - it will corrupt the files or not work at all. Back when I started translating Serpent Isle in 2015 Andrea was kind enough to patch his latest version 0.98 to "0.98SE". This version is good for SI AND BG.

This said, please allow me to share my best practices to work most efficiently:

- Copy U7ED to an empty folder of choice
- Initially startup U7ED and configure it
- Choose the first dialogue in the drop down field
- Use your DOWN cursor to highlight all following +700 dialogues (highlighting them will make U7ED creating them in the folder U7ED resides in)
- Once you are done, you have all dialogues as HEXA.txt in the folder which you can now modify with a text-processor of your choice.
- After that, (only) use U7ed to compile your modified texts.

Best of luck!

John

@DominusExult it might make sense to only go for 0.98SE since this version is compatible with both parts of U7. Upload or mail? (1.4m)

**Reply by DominusExult** [reply]
Thanks @ungesundes-halbwissen! Best upload it here until I find a proper place to put it. If I only have it per mail, I'm likely to forget it ;)

**Reply by r-h-o-n-a** [reply]
@ungesundes-halbwissen thanks very much for that! Yes, I'd definitely found that 0.94 had a tendency to corrupt the extracted .txt files occasionally when working with SI, and that if I didn't delete those corrupted .txt files before recompiling then it'd bork the entire game, though I'd been successfully skirting the issue up to now by vigilantly policing the files. I wasn't aware that a newer version had been successfully implemented and it'll be a great relief to be able to work with 0.98SE instead - many thanks for uploading the new version, and my gratitude also to Andrea M., for implementing it!

**Reply by r-h-o-n-a** [reply]
@DominusExult I'm presuming that @ungesundes-halbwissen having uploaded the newer version will obviate the need for me to provide the older form of the utility, although I'm happy to send it on for archival purposes if you want :)

The translation I'm doing is into Ubykh, a recently extinct language of Turkey. I'm a linguist working on a comprehensive dictionary of this language, and also an old Ultima fiend from way back when I was a kid. SI was always my favourite, and basically, my hope is eventually to use the Ubykh translation of SI as a language teaching tool. Because of the incredibly rich implementation of the environments in U7, plus the ability to click on any object and be given its name, my hope is to use SI as a sort of virtual world in which you can learn what objects are called, as well as learning both simple and complex sentence structures. (Ultima lends itself well to that, with its highly repetitive "name - job - bye" pattern that should be good for reinforcing formulaic phrases.) The SI world is also fairly close to the pre-industrial nature of traditional Ubykh society before they were ethnically cleansed from the Caucasus, and many recorded Ubykh folk tales include magic and mythological themes (dragons, sorcery, ghosts, a few different types of humanoid cryptids and such), so almost everything in the SI world can be expressed pretty easily in Ubykh.

No website or repository as yet, as the progress is pretty slow (I'm long-term unemployed and managing some serious mental health issues), but that's definitely a goal at some point. I'm pretty passionate about Ubykh in general, and an SI translation in particular I think could be a powerful - and fun - tool to eventually help revive the language.

**Reply by DominusExult** [reply]
@r-h-o-n-a, yes, no need for the older version (unless there is any readme file or any other file included).

Keep us posted on your translation. The reason why you are doing this is very interesting and I would like to follow its progress.

**Reply by ungesundes-halbwissen** [reply]
There are several additional files in the original package, one directory called help contains an htm page and pngs. So chances are: yes ;)

Available for download here: http://ultima.oldgamesitalia.net/?p=download

**Comment by ungesundes-halbwissen** [comment]
[_U7Ed-SI.zip](https://github.com/exult/exult/files/10445706/_U7Ed-SI.zip)

U7-Editor v0.98SE - working version for Black Gate and Serpent Isle (patched 2015 by the author Andrea M. to work on SI and with accented chars)
